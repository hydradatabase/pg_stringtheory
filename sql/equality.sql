CREATE EXTENSION IF NOT EXISTS stringtheory;

-- no match
SELECT stringtheory.equals('hello', 'world');

-- match
SELECT stringtheory.equals('hello', 'hello');

-- match on a 16 byte boundary
SELECT stringtheory.equals('1234567890123456', '1234567890123456');

-- no match when partial
SELECT stringtheory.equals('123456', '12345');

-- test equality in a CTE
WITH a AS (SELECT md5(generate_series(1, 1000)::text) b)
SELECT count(*) FROM a
WHERE stringtheory.equals(b, md5('123'));

-- test strstr in a table
CREATE TEMPORARY TABLE stringtheory_test
(a text);

INSERT INTO stringtheory_test
SELECT md5(generate_series(1,1000)::text);

SELECT COUNT(*) FROM stringtheory_test
WHERE stringtheory.equals(a, md5('123'));

DROP TABLE stringtheory_test;
