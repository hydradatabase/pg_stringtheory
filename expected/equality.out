CREATE EXTENSION IF NOT EXISTS stringtheory;
-- no match
SELECT stringtheory.equals('hello', 'world');
 equals 
--------
 f
(1 row)

-- match
SELECT stringtheory.equals('hello', 'hello');
 equals 
--------
 t
(1 row)

-- match on a 16 byte boundary
SELECT stringtheory.equals('1234567890123456', '1234567890123456');
 equals 
--------
 t
(1 row)

-- no match when partial
SELECT stringtheory.equals('123456', '12345');
 equals 
--------
 f
(1 row)

-- test equality in a CTE
WITH a AS (SELECT md5(generate_series(1, 1000)::text) b)
SELECT count(*) FROM a
WHERE stringtheory.equals(b, md5('123'));
 count 
-------
     1
(1 row)

-- test strstr in a table
CREATE TEMPORARY TABLE stringtheory_test
(a text);
INSERT INTO stringtheory_test
SELECT md5(generate_series(1,1000)::text);
SELECT COUNT(*) FROM stringtheory_test
WHERE stringtheory.equals(a, md5('123'));
 count 
-------
     1
(1 row)

DROP TABLE stringtheory_test;
